### PR TITLE
[aggregator] Use a single goroutine to emit metrics for all TCP client queues

### PR DESCRIPTION
**Pull request by vdarulis**
**What this PR does / why we need it**:

It's less than ideal to have 2 extra goroutines per aggregator instance, as it adds scheduling overhead. This PR cuts it down to 1.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/aggregator/client/queue.go [hunk 1]
```diff
@@ -101,6 +95,9 @@ type instanceQueue interface {
 	// Enqueue enqueues a data buffer.
 	Enqueue(buf protobuf.Buffer) error
 
+	// Size returns the number of times in the queue.
```
Comment (mway) [inline src/aggregator/client/queue.go:98]: s/times/items/

File: src/aggregator/client/queue.go [hunk 2]
```diff
@@ -154,9 +151,8 @@ func newInstanceQueue(instance placement.Instance, opts Options) instanceQueue {
 	}
 	q.writeFn = q.conn.Write
 
-	q.wg.Add(2)
+	q.wg.Add(1)
```
Comment (mway) [inline src/aggregator/client/queue.go:154]: while we're here, can you move the wg decr to here:

```go
q.wg.Add(1)
go func() {
  defer q.wg.Done()
  q.drain()
}()
```

File: src/aggregator/client/writer_mgr_test.go [hunk 3]
```diff
@@ -258,9 +266,12 @@ func TestWriterManagerCloseSuccess(t *testing.T) {
...
 			wr := w.instanceWriter.(*writer)
-			wr.Lock()
-			defer wr.Unlock()
-			if !wr.closed {
+			closed := func() bool {
+				wr.Lock()
+				defer wr.Unlock()
+				return wr.closed
+			}
+			if !closed() {
```
Comment (mway) [inline src/aggregator/client/writer_mgr_test.go:274]: nit: closure is overkill

```go
wr.Lock()
closed := wr.closed
wr.Unlock()

if !closed {
  return false
}
```

**Review by mway — APPROVED** [review]

File: src/aggregator/client/writer_mgr_test.go (hunk 3 above)
Comment (vdarulis) [reply src/aggregator/client/writer_mgr_test.go:274]: whoops, refactoring leftoever. thanks

**Pull request merged**